### ISSUE TITLE
Warning: Using UNSAFE_componentWillReceiveProps in strict mode is not recommended and may indicate bugs in your code

### ISSUE THREAD

**Issue by JohnTheBeloved** [alex3165/react-mapbox-gl#872]
Warning: Using UNSAFE_componentWillReceiveProps in strict mode is not recommended and may indicate bugs in your code. See https://fb.me/react-unsafe-component-lifecycles for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state

Please update the following components: ReactMapboxGl

In addition to the warning, The map panning and zooming appears broken

React version: ^16.13.1
ReactMapboxGl version: ^4.8.6

**Comment by macduff111** [comment]
I am getting the same warnings for ReactMapboxGl, Layer, EnhancedLayer components, but map pan and zoom working OK

**Comment by wkolcz-UMHS** [comment]
I am getting the same issue for ReactMapboxGl and GeoJSONLayer but it all works ok so far

**Comment by phuquoc164** [comment]
I am getting the same issue for ReactMapboxGl, ScaleControl and Layer but it all works

**Comment by aasutossh** [comment]
Any news on this?

**Comment by avillarubia** [comment]
anyone can save us please?

**Comment by clairefro** [comment]
Same - this didn't used to happen. A quick search of the `react-mapbox-gl` code didn't reveal any use of `componentWillReceiveProps`, so maybe it is from one of the dependencies?

**Comment by avillarubia** [comment]
@clairefro yes, A big chances it is from them. Hoping this will be resolved asap.

**Comment by avillarubia** [comment]
@phuquoc164 @wkolcz-UMHS how did you do that? Can you show us please?

**Comment by clairefro** [comment]
i ended up switching to `react-map-gl` as I was in a time crunch...

**Comment by avillarubia** [comment]
@clairefro thanks you, I will try that

**Comment by phuquoc164** [comment]
@avillarubia I just use this library normally. For the warnings, you can delete <React.StrictMode></React.StrictMode> in index.tsx
